### PR TITLE
[IoTDB-1936] Support other aggregations in group by fill with value filter

### PR DESCRIPTION
**Pull request by CRZbulabula**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+      initExtraArrays(extraPreviousValues, extraPreviousTimes, true, extraPreviousGenerators);
+    }
+    if (extraNextGenerators != null) {
+      initExtraArrays(extraNextValues, extraNextTimes, false, extraNextGenerators);
+    }
+
+    initCachedTimesAndValues();
+  }
+
+  private void initPathGenerators(QueryContext context, GroupByTimeFillPlan groupByTimeFillPlan)
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java:94]: Why is the method called `initPathGenerators`? I think it should be renamed to `initReadersByTimestamp()`

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    try {
+      replaceGroupByFilter(newExpression, timeFilter);
+    } catch (IllegalPathException ignore) {
+      // ignored
+    }
+    return newExpression;
+  }
+
+  /* Init extra path executors to query data outside the original group by query */
+  private void initExtraGenerators(QueryContext context, GroupByTimeFillPlan groupByTimeFillPlan)
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java:176]: rename it to `initExtraReadersByTimestamp`

**Review by JackieTien97 — APPROVED** [review]

**Review by JackieTien97 — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java (hunk 1 above)
Comment (CRZbulabula) [reply server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java]: done

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java (hunk 2 above)
Comment (CRZbulabula) [reply server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillWithValueFilterDataSet.java]: done

**Pull request merged**